doc/src/challenges/0004-btree-node.md

```diff
@@ -0,0 +1,67 @@
+# Challenge 4: Towards the memory safety of BTreeMap
+
+- **Status:** Open
+- **Tracking Issue:** [Link to issue](https://github.com/model-checking/verify-rust-std/issues/25)
+- **Start date:** *2024-07-01*
+- **End date:** *2024-12-10*
+
+-------------------
+
+
+## Goal
+
+Verify the memory safety of the [`alloc::collections::btree::node` module](https://github.com/rust-lang/rust/blob/c290e9de32e8ba6a673ef125fde40eadd395d170/library/alloc/src/collections/btree/node.rs).
+This is one of the main modules used for implementing the `BTreeMap` collection, and it includes a lot of unsafe code.
+
+### Success Criteria
+
+The memory safety of all public functions (especially safe ones) containing unsafe code must be verified, e.g.:
+
+1. `LeafNode::new`
```

Comment:
nit: Can you change this to a unordered list or to an ordered number list? right now, this appears as a list with all indices being marked as 1.

Reply:
Do you mean it appears with all 1's in the rendered version? The numbering in the GitHub version looks correct: https://github.com/zhassan-aws/verify-rust-std/blob/btree-node/doc/src/challenges/0006-btree-node.md
But maybe the rendering doesn't behave the same?

Reply:
Or you meant the issue is the source text itself?